src/one_dragon/envs/git_service.py

```
@@ -47,12 +78,30 @@ def __init__(
         self._progress_callback: Callable[[float, str], None] = progress_callback
         self._last_message: str | None = None
         self._last_log_at: float | None = None
+        self._first_transfer: bool = True
+
+    def sideband_progress(self, string: str) -> None:
+        """远端 git 的实时输出（如 "Compressing objects", "Counting objects"）。"""
+        # 空行/空白仅作换行占位，不转发
+        if not string or not string.strip():
+            return
+        text = string.strip()
+        # 过滤纯进度百分比行（如 "remote: 20% (100/500)"），
```

Ruff (0.15.20)

[warning] Comment contains ambiguous `（` (FULLWIDTH LEFT PARENTHESIS). Did you mean `(` (LEFT PARENTHESIS)?

(RUF003)